Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -1262,7 +1263,43 @@ public unsafe bool WasPressedThisFrame()
             {
                 var actionStatePtr = &state.actionStates[m_ActionIndexInState];
                 var currentUpdateStep = InputUpdate.s_UpdateStepCount;
-                return actionStatePtr->pressedInUpdate == currentUpdateStep && currentUpdateStep != default && actionStatePtr->frame == ExpectedFrame();
+                return actionStatePtr->pressedInUpdate == currentUpdateStep && currentUpdateStep != default;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if the action's value crossed the press threshold (see <see cref="InputSettings.defaultButtonPressPoint"/>)
+        /// in the MonoBehaviour Update cycle (rendering frame).
+        /// </summary>
+        /// <returns>True if the action was pressed in the MonoBehaviour Update cycle (rendering frame).</returns>
+        /// <remarks>
+        /// Unlike <see cref="WasPressedThisFrame"/>, this method will return true only if the InputSystem was updated and the action was pressed in the current dynamic Update cycle (in between the previous and the current frame).
+        /// This can be used in dynamic update if the <see cref="InputSettings.updateMode"/> is set to <see cref="InputSettings.UpdateMode.ProcessEventsInFixedUpdate"/> or <see cref="InputSettings.UpdateMode.ProcessEventsManually"/>.
+        /// If the update mode is set to <see cref="InputSettings.UpdateMode.ProcessEventsInDynamicUpdate"/>, this method will behave exactly like <see cref="WasPressedThisFrame"/>.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// var fire = playerInput.actions["fire"];
+        /// if (fire.WasPressed() &amp;&amp; fire.IsPressed())
+        ///     StartFiring();
+        /// else if (fire.WasReleasedThisFrame())
+        ///     StopFiring();
```

Comment:
```suggestion
        /// if (fire.WasPressedThisDynamicUpdate() &amp;&amp; fire.IsPressed())
        ///     StartFiring();
        /// else if (fire.WasReleasedThisDynamicUpdate())
        ///     StopFiring();
```
Copy-paste mistake, this code sample for this method doesn't use the `WasPressedThisDynamicUpdate` method it's for.